openupgrade_scripts/scripts/hr_holidays/15.0.1.5/pre-migration.py

```diff
@@ -0,0 +1,240 @@
+from openupgradelib import openupgrade
+
+
+def _fast_fill_hr_leave_employee_company_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE hr_leave
+        ADD COLUMN IF NOT EXISTS employee_company_id integer""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE hr_leave hl
+        SET employee_company_id = empl.company_id
+        FROM hr_employee empl
+        WHERE empl.company_id IS NOT NULL AND hl.employee_id = empl.id""",
+    )
+
+
+def _map_hr_leave_state(env):
+    openupgrade.map_values(
```

Comment:
Please copy the column here in pre-migration, and do the map on post-migration. This way, we preserve old values for any possible extra migration script (imagine there's an extension module that was putting an information if cancel - if we don't copy the column, we lose where we should see -).